patches/0004-desktop-editor-intigration.patch

```diff
@@ -0,0 +1,209 @@
+-      return path
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    for (const path of paths) {
++      console.log(path)
++      if (path.indexOf("/usr/") > -1) {
++        if (await pathExists("/var/run/host".toString().concat(path))) {
++          return path
++        }
++      } else {
++        if (await pathExists(path)) {
++          return path
++        }
++      }
++    }
++  } else {
++    for (const path of paths) {
++      if (await pathExists(path)) {
++        return path
++      }
+     }
+   }
+-
+   return null
+ }
+ 
+diff --git a/app/src/lib/shells/linux.ts b/app/src/lib/shells/linux.ts
+index 47a5c7f03..a73a9ffa8 100644
+--- a/app/src/lib/shells/linux.ts
++++ b/app/src/lib/shells/linux.ts
+@@ -27,7 +27,12 @@ export function parse(label: string): Shell {
+ }
+ 
+ async function getPathIfAvailable(path: string): Promise<string | null> {
+-  return (await pathExists(path)) ? path : null
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    var path_to_search: string = "/var/run/host".toString().concat(path)
++    return (await pathExists(path_to_search)) ? path : null
++  } else {
++    return (await pathExists(path)) ? path : null
++  }
+ }
+ 
+ function getShellPath(shell: Shell): Promise<string | null> {
+@@ -157,29 +162,62 @@ export function launch(
+   path: string
+ ): ChildProcess {
+   const shell = foundShell.shell
+-  switch (shell) {
+-    case Shell.Gnome:
+-    case Shell.Mate:
+-    case Shell.Tilix:
+-    case Shell.Terminator:
+-    case Shell.XFCE:
+-    case Shell.Alacritty:
+-      return spawn(foundShell.path, ['--working-directory', path])
+-    case Shell.Urxvt:
+-      return spawn(foundShell.path, ['-cd', path])
+-    case Shell.Konsole:
+-      return spawn(foundShell.path, ['--workdir', path])
+-    case Shell.Xterm:
+-      return spawn(foundShell.path, ['-e', '/bin/bash'], { cwd: path })
+-    case Shell.Terminology:
+-      return spawn(foundShell.path, ['-d', path])
+-    case Shell.Deepin:
+-      return spawn(foundShell.path, ['-w', path])
+-    case Shell.Elementary:
+-      return spawn(foundShell.path, ['-w', path])
+-    case Shell.Kitty:
+-      return spawn(foundShell.path, ['--single-instance', '--directory', path])
+-    default:
+-      return assertNever(shell, `Unknown shell: ${shell}`)
++  if (process.env.FLATPAK_HOST !== 'undefined') {
++    switch (shell) {
++      case Shell.Gnome:
++        return spawn("flatpak-spawn", ["--host", foundShell.path])
++      case Shell.Mate:
++        return spawn("flatpak-spawn", ["--host", foundShell.path])
++      case Shell.Tilix:
++        return spawn("flatpak-spawn", ["--host", foundShell.path])
++      case Shell.Terminator:
++        return spawn("flatpak-spawn", ["--host", foundShell.path])
++      case Shell.XFCE:
++        return spawn("flatpak-spawn", ["--host", foundShell.path])
++      case Shell.Alacritty:
++        return spawn("flatpak-spawn", ["--host", foundShell.path, '--working-directory', path])
```

Comment:
🤔 not sure if this was intended, but the `switch` statement mans these cases can all fall through to the `Alacritty` example, which matches the original code and ensures `--working-directory` passes in the user's repository:

```suggestion
+      case Shell.Gnome:
+      case Shell.Mate:
+      case Shell.Tilix:
+      case Shell.Terminator:
+      case Shell.XFCE:
+      case Shell.Alacritty:
+        return spawn("flatpak-spawn", ["--host", foundShell.path, '--working-directory', path])
```

Reply:
I'm using flatpak-spawn since it launches out side the sandbox, if you have any ideas on  how to do it without this, I'm game